nsqlbproducer/lbproducer.go

```diff
@@ -0,0 +1,94 @@
+package nsqlbproducer
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/Scalingo/go-utils/nsqproducer"
+	nsq "github.com/nsqio/go-nsq"
+	errgo "gopkg.in/errgo.v1"
+)
+
+type NsqLBProducer struct {
+	producers []nsqproducer.Producer
+	randSrc   randSource
+}
+
+type Host struct {
+	Host string
+	Port string
+}
+
+type LBProducerOpts struct {
+	Hosts      []Host
+	NsqConfig  *nsq.Config
+	SkipLogSet map[string]bool
+}
+
+type randSource interface {
+	Int() int
+}
+
+func New(opts LBProducerOpts) (*NsqLBProducer, error) {
+	if len(opts.Hosts) == 0 {
+		return nil, fmt.Errorf("A producer must have at least one host")
+	}
+	producer := &NsqLBProducer{
+		producers: make([]nsqproducer.Producer, len(opts.Hosts)),
+	}
+
+	for i, h := range opts.Hosts {
+		p, err := nsqproducer.New(nsqproducer.ProducerOpts{
+			Host:       h.Host,
+			Port:       h.Port,
+			NsqConfig:  opts.NsqConfig,
+			SkipLogSet: opts.SkipLogSet,
+		})
+
+		if err != nil {
+			return nil, errgo.Mask(err)
+		}
+
+		producer.producers[i] = p
+	}
+
+	producer.randSrc = rand.New(rand.NewSource(time.Now().Unix()))
+
+	return producer, nil
+}
+
+func (p *NsqLBProducer) Publish(ctx context.Context, topic string, message nsqproducer.NsqMessageSerialize) error {
+	firstProducer := p.randSrc.Int() % len(p.producers)
+
+	var err error
+	for i := 0; i < len(p.producers); i++ {
+		err = p.producers[(i+firstProducer)%len(p.producers)].Publish(ctx, topic, message)
```

Comment:
Do we really want to ignore the errors? I mean if there are multiple producers and one fail, we will not be notified about this error

Reply:
Yeah that's the point, if a a producer get the message it's good enough.
Monitoring the cluster health is another issue which is not in the scope of this PR IMHO.

Reply:
Do you now what are the reason for a failed publish? How could we monitor such problem elsewhere than with this error?

Reply:
A node is down (or is being upgraded) we could monitor it with Philae (having one probe per member of the cluster).

Reply:
Why not. I am not a big fan of ignoring such errors but OK as it didn't hurt our beloved CTO

Reply:
What about logging all the errors?

Reply:
The most common case of this error will be node reboot, nsq reboot, and nsq update. Which are not critical nor important. The only important thing is that the message is sent. So if all node do not respond thats bad, but if one node does not we don't care.

Reply:
I do not like it here since this is a library, logging might be hard since we do not know how the user might format his logs.

Reply:
You meant logrus.Logger ?

Reply:
logrus.FieldLogger actually